### PR TITLE
Changes to support micronaut-core

### PR DESCRIPTION
**Pull request by alvarosanchez**
- Correctly apply shadow configurations.
- Honour extra pom information from parent and bom projects

### CODE REVIEW

**Comment by graemerocher** [comment]
Is `shadowJarEnabled` defined somewhere?

File: src/main/groovy/io/micronaut/build/MicronautPublishingPlugin.groovy [hunk 1]
```diff
@@ -63,126 +67,135 @@ class MicronautPublishingPlugin implements Plugin<Project> {
...
+
+                    publications {
+                        if (project.extensions.findByType(PublishingExtension).publications.empty) {
+                            maven(MavenPublication) { publication ->
+                                artifactId( "micronaut-" + project.name.substring(project.name.indexOf('/') + 1) )
+
+                                if(project.shadowJarEnabled == true) {
+                                    def shadowJar = tasks.findByName("shadowJar")
+                                    artifact(project.tasks.shadowJar) {
+                                        classifier = null
```
Comment (graemerocher) [inline src/main/groovy/io/micronaut/build/MicronautPublishingPlugin.groovy:112]: I believe `classifier` is deprecated and you have to use `artefactClassifier.set(null)` in newer versions of Gradle

**Comment by alvarosanchez** [comment]
> Is `shadowJarEnabled` defined somewhere?

In all core projects. I have guarded property evaluation with a `hasProperty` check.

File: src/main/groovy/io/micronaut/build/MicronautPublishingPlugin.groovy (hunk 1 above)
Comment (alvarosanchez) [reply src/main/groovy/io/micronaut/build/MicronautPublishingPlugin.groovy:112]: It is not deprecated nor it produces any deprecation warning.

As far as I can tell, `classifier = null` will invoke [this method](https://docs.gradle.org/current/javadoc/org/gradle/api/publish/maven/MavenArtifact.html#setClassifier-java.lang.String-) which is not deprecated.

**Comment by alvarosanchez** [comment]
> Can you run this and make sure it doesn't cause any deprecation warnings for Gradle 7?

The only warnings produced are from `shadowJar`:

```
> Task :aop:shadowJar
Property 'dependencyFilter' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.5/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Property 'internalCompressor' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.5/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Property 'relocators' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.5/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Property 'rootPatternSet' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.5/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Property 'stats' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.5/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Property 'transformers' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.5/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
```

**Comment by graemerocher** [comment]
Those warnings must be an out of date shadow plugin being used

**Comment by alvarosanchez** [comment]
Found the culprit:

```
+--- io.micronaut.docs:micronaut-docs-gradle-plugins:1.0.23
|    +--- org.codehaus.groovy:groovy:2.5.4 -> 3.0.3
|    +--- com.github.jengelman.gradle.plugins:shadow:2.0.2
```

Going to exclude that

**Pull request merged**